cpp/include/cudf/io/csv.hpp

```diff
@@ -0,0 +1,1447 @@
+
+  /**
+   * @brief Whether to keep the built-in default NA values.
+   */
+  bool is_enabled_keep_default_na() const { return _keep_default_na; }
+
+  /**
+   * @brief Whether to disable null filter.
+   */
+  bool is_enabled_na_filter() const { return _na_filter; }
+
+  /**
+   * @brief Whether to parse dates as DD/MM versus MM/DD.
+   */
+  bool is_enabled_dayfirst() const { return _dayfirst; }
+
+  /**
+   * @brief Returns timestamp_type to which all timestamp columns will be cast.
+   */
+  data_type get_timestamp_type() const { return _timestamp_type; }
+
+  /**
+   * @brief Sets compression format of the source.
+   *
+   * @param comp Compression type.
+   */
+  void set_compression(compression_type comp) { _compression = comp; }
+
+  /**
+   * @brief Sets number of bytes to skip from source start.
+   *
+   * @param offset Number of bytes of offset.
+   */
+  void set_byte_range_offset(size_type offset)
+  {
+    if ((offset != 0) and ((_skiprows != 0) or (_skipfooter != 0) or (_nrows != -1))) {
+      CUDF_FAIL(
+        "When ther is valid value in skiprows or skipfooter or nrows, offset can't have non-zero "
+        "value");
+    }
+    _byte_range_offset = offset;
+  }
+
+  /**
+   * @brief Sets number of bytes to read.
+   *
+   * @param size Number of bytes to read.
+   */
+  void set_byte_range_size(size_type size)
+  {
+    if ((size != 0) and ((_skiprows != 0) or (_skipfooter != 0) or (_nrows != -1))) {
+      CUDF_FAIL(
+        "When ther is valid value in skiprows or skipfooter or nrows, range size can't have "
+        "non-zero value");
+    }
+    _byte_range_size = size;
+  }
+
+  /**
+   * @brief Sets names of the column.
+   *
+   * @param col_names Vector of column names.
+   */
+  void set_names(std::vector<std::string> col_names) { _names = std::move(col_names); }
+
+  /**
+   * @brief Sets prefix to be used for column ID.
+   *
+   * @param String used as prefix in for each column name.
+   */
+  void set_prefix(std::string pfx) { _prefix = pfx; }
+
+  /**
+   * @brief Sets whether to rename duplicate column names.
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_mangle_dupe_cols(bool val) { _mangle_dupe_cols = val; }
+
+  /**
+   * @brief Sets names of the columns to be read.
+   *
+   * @param col_names Vector of column names that are needed.
+   */
+  void set_use_cols_names(std::vector<std::string> col_names)
+  {
+    _use_cols_names = std::move(col_names);
+  }
+
+  /**
+   * @brief Sets indexes of columns to read.
+   *
+   * @param col_ind Vector of column indices that are needed.
+   */
+  void set_use_cols_indexes(std::vector<int> col_ind) { _use_cols_indexes = std::move(col_ind); }
+
+  /**
+   * @brief Sets number of rows to read.
+   *
+   * @param val Number of rows to read.
+   */
+  void set_nrows(size_type val)
+  {
+    CUDF_EXPECTS((val == 0) or (_nrows == -1), "Cannot use both `_nrows` and `_skipfooter`");
```

Comment:
copypasta error here
```suggestion
    CUDF_EXPECTS((val == 0) or (_skipfooter == 0), "Cannot use both `nrows` and `skipfooter`");
```